service/src/main/java/bio/terra/profile/service/iam/SamService.java

```diff
@@ -345,6 +352,71 @@ public void deleteProfileResource(AuthenticatedUserRequest userRequest, UUID pro
     }
   }
 
+  /**
+   * Deletes the managed resource group associated with the given billing profile in Sam.
+   *
+   * @param profileId profile ID whose MRG entry in Sam should be deleted
+   * @param userRequest authenticated user
+   * @throws InterruptedException
+   */
+  public void deleteManagedResourceGroup(UUID profileId, AuthenticatedUserRequest userRequest)
+      throws InterruptedException {
+    AzureApi azureApi = samAzureApi(userRequest.getToken());
+    try {
+      SamRetry.retry(() -> azureApi.deleteManagedResourceGroup(profileId.toString()));
+      logger.info("Deleted mrg in Sam for profile {}", profileId);
+    } catch (ApiException e) {
+      if (e.getCode() == HttpStatus.NOT_FOUND.value()) {
+        // MRG may already be deleted or not present
+        return;
+      }
+      throw SamExceptionFactory.create(
+          "Error deleting Sam managed resource group record for billing profile", e);
+    }
+  }
+
+  /**
+   * Creates a record in Sam that links a billing profile to an Azure MRG
+   *
+   * @param profile Billing profile that will be linked
+   * @param userRequest authenticated user
+   * @throws InterruptedException
+   */
+  public void createManagedResourceGroup(
+      BillingProfile profile, AuthenticatedUserRequest userRequest) throws InterruptedException {
+    AzureApi azureApi = samAzureApi(userRequest.getToken());
+    try {
+      SamRetry.retry(
+          () ->
+              azureApi.createManagedResourceGroup(
+                  profile.id().toString(),
+                  new ManagedResourceGroupCoordinates()
+                      .tenantId(profile.getRequiredTenantId().toString())
+                      .subscriptionId(profile.getRequiredSubscriptionId().toString())
+                      .managedResourceGroupName(profile.getRequiredManagedResourceGroupId())));
+
+      logger.info(
+          "Created mrg in Sam for profile {}, mrg id = {}, tenant = {}, subscription = {}",
+          profile.id(),
+          profile.managedResourceGroupId(),
+          profile.tenantId(),
+          profile.subscriptionId());
+    } catch (ApiException e) {
+      if (e.getCode() == HttpStatus.CONFLICT.value()) {
+        // MRG has already been created in Sam, move on
+        logger.info(
+            "Mrg record already exists in Sam for profile {},  mrg id = {}, tenant = {}, subscription = {}",
+            profile.id(),
+            profile.managedResourceGroupId(),
+            profile.tenantId(),
+            profile.subscriptionId());
+        return;
```

Comment:
Is `CONFLICT` returned when there's a record in Sam where the profile.id is there, but the MRG coordinates don't match, or vice versa? 
I would think we'd want to throw the exception unless the record in Sam matches exactly (in which case this is essentially a no-op), and otherwise return an error that the MRG is already assigned.